app/src/components/form/attribute_input.rs

```diff
@@ -0,0 +1,193 @@
+use crate::{
+    components::form::{date_input::DateTimeInput, file_input::JpegFileInput},
+    infra::{schema::AttributeType, tooltip::Tooltip},
+};
+use web_sys::Element;
+use yew::{
+    function_component, html, use_effect_with_deps, use_mut_ref, use_node_ref,
+    virtual_dom::AttrValue, Component, Context, Html, Properties,
+};
+
+#[derive(Properties, PartialEq)]
+struct AttributeInputProps {
+    name: AttrValue,
+    attribute_type: AttributeType,
+    #[prop_or(None)]
+    value: Option<String>,
+}
+
+#[function_component(AttributeInput)]
+fn attribute_input(props: &AttributeInputProps) -> Html {
+    let input_type = match props.attribute_type {
+        AttributeType::String => "text",
+        AttributeType::Integer => "number",
+        AttributeType::DateTime => {
+            return html! {
+                <DateTimeInput name={props.name.clone()} value={props.value.clone()} />
+            }
+        }
+        AttributeType::Jpeg => {
+            return html! {
+                <JpegFileInput name={props.name.clone()} value={props.value.clone()} />
+            }
+        }
+    };
+
+    html! {
+        <input
+            type={input_type}
+            name={props.name.clone()}
+            class="form-control"
+            value={props.value.clone()} />
+    }
+}
+
+#[derive(Properties, PartialEq)]
+struct AttributeLabelProps {
+    pub name: String,
+}
+#[function_component(AttributeLabel)]
+fn attribute_label(props: &AttributeLabelProps) -> Html {
+    let tooltip_ref = use_node_ref();
+    let tooltip_js = use_mut_ref(|| None);
+
+    {
+        let tooltip_js = tooltip_js.clone();
+        use_effect_with_deps(
+            move |tooltip_ref| {
+                let tooltip_ref = tooltip_ref
+                    .cast::<Element>()
+                    .expect("Tooltip element should exist");
+                *tooltip_js.borrow_mut() = Some(Tooltip::new(tooltip_ref));
+                || {}
+            },
+            tooltip_ref.clone(),
+        );
+    }
```

Comment:
We actually don't need `tooltip_js` (value is never used; was only used in aforementioned commented-out code).

So you can probably simplify this to:
```suggestion
    use_effect_with_deps(
        move |tooltip_ref| {
            let tooltip_ref = tooltip_ref
                .cast::<Element>()
                .expect("Tooltip element should exist");
            Tooltip::new(tooltip_ref);
            || {}
        },
        tooltip_ref.clone(),
    );
```

Yay less magic! (And do note that using `use_node_ref` with `use_effect_with_deps` is suggested by [Yew's documentation](https://yew.rs/docs/0.19.0/concepts/function-components/pre-defined-hooks#use_node_ref))

Reply:
Thanks!